core/common/src/main/java/alluxio/security/authentication/SaslStreamClientDriver.java

```diff
@@ -88,43 +93,44 @@ public void setServerObserver(StreamObserver<SaslMessage> requestObserver) {
   @Override
   public void onNext(SaslMessage saslMessage) {
     try {
-      LOG.debug("SaslClientDriver received message: {}",
-          saslMessage != null ? saslMessage.getMessageType().toString() : "<NULL>");
+      LOG.debug("SaslClientDriver received message: {} for channel: {}", saslMessage, mChannelId);
       SaslMessage response = mSaslHandshakeClientHandler.handleSaslMessage(saslMessage);
       if (response != null) {
         mRequestObserver.onNext(response);
       } else {
         // {@code null} response means server message was a success.
         mHandshakeFuture.set(true);
       }
-    } catch (SaslException e) {
+    } catch (Exception e) {
+      LOG.debug("Exception while handling SASL message: {} for channel: {}. Error: {}", saslMessage,
+          mChannelId, e);
       mHandshakeFuture.setException(e);
-      mRequestObserver
-          .onError(Status.fromCode(Status.Code.UNAUTHENTICATED).withCause(e).asException());
+      mRequestObserver.onError(e);
     }
   }
 
   @Override
   public void onError(Throwable throwable) {
+    LOG.warn("Received error on client driver for channel: {}. Error: {}", mChannelId, throwable);
```

Comment:
Reverting to WARN, here and at `SaslStreamServerDriver` as it proved very useful in a recent error investigation.